File: src/types/history.ts


export interface ScanHistoryEntry {
    id: string;
    label: string;
    timestamp: string;
    drive_path: string;
    total_size: number;
    file_count: number;
}

export interface ScanDiff {
    added: string[];
    removed: string[];
    grown: [string, number, number][]; // path, old_size, new_size
    shrunk: [string, number, number][];
}

File: src-tauri/src/models/history.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ScanHistoryEntry {
    pub id: String,
    pub label: String,
    pub timestamp: String,
    pub drive_path: String,
    pub total_size: u64,
    pub file_count: u64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ScanDiff {
    pub added: Vec<String>,
    pub removed: Vec<String>,
    pub grown: Vec<(String, u64, u64)>, // path, old_size, new_size
    pub shrunk: Vec<(String, u64, u64)>, 
}
